### ISSUE TITLE
Where's the Javascript client API documentation?

### ISSUE THREAD

**Issue by karlrwjohnson** [aspnet/Docs#7015]
What methods are available for the HubConnection class?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3cf283c9-5cd2-a9ff-b8c8-372d8a04356d
* Version Independent ID: 729fddcf-24e4-b563-4952-a8045b231868
* Content: [ASP.NET Core SignalR JavaScript client](https://docs.microsoft.com/en-us/aspnet/core/signalr/javascript-client?view=aspnetcore-2.1#feedback)
* Content Source: [aspnetcore/signalr/javascript-client.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/signalr/javascript-client.md)
* Product: **aspnet-core**
* GitHub Login: @rachelappel
* Microsoft Alias: **rachelap**

**Comment by msimecek** [comment]
I didn't find any documentation describing it, but at least the source code is quite straight-forward: https://github.com/aspnet/SignalR/blob/dev/clients/ts/signalr/src/HubConnection.ts

**Comment by rachelappel** [comment]
* Since the Hubconnection is a server-side component, it is not in the client-side JS doc. The API reference for it is here: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr?view=aspnetcore-2.1

* A table with methods from the server-side hub you are looking for are here in the hubs doc: https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.1

* The starting point for the API reference documentation that details a list of methods for hubs in Core SignalR is here: https://docs.microsoft.com/en-us/dotnet/api/?view=aspnetcore-2.1

* The link you provided is the old version of SignalR for .NET, not ASP.NET Core SignalR and some of that is incompatible with current versions

**Comment by msimecek** [comment]
@rachelappel Could you please clarify how is HubConnection not a client-side JS component? I'm looking at the client library (@aspnet/signalr which is referenced in this document) and there is definitely `HubConnection` object produced by the `HubConnectionBuilder.build()`. My link shows exactly that class, written in TypeScript, for browser client... It's not .NET.

**Comment by rachelappel** [comment]
@msimecek My bad, I conflated this issue with something open in a separate tab, similar, about hubs, which caused me to think you meant Hub when writing HubConnection (also a typo where I typed out the full word Hubconnection & meant hub). Sorry for the confusion there. 

The most common methods are On, invoke, and Start as well as creating the instance of the connection itself. A goal was to make it so that folks only need those few methods for most app scenarios. The Js doc has some detail about the most used methods, but not a complete list. A doc on configuration (of the connection and general signalr capabilities) that is soon to be published will have some more details about HubConnection and more JS code.
